modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/robotic/synthetic.dm

```diff
@@ -0,0 +1,165 @@
+	)
+	changesource_flags = MIRROR_BADMIN | WABBAJACK | MIRROR_MAGIC | MIRROR_PRIDE | ERT_SPAWN | RACE_SWAP | SLIME_EXTRACT
+	reagent_flags = PROCESS_SYNTHETIC
+	burnmod = 1.3 // Every 0.1% is 10% above the base.
+	brutemod = 1.3
+	coldmod = 1.2
+	heatmod = 2
+	siemens_coeff = 1.4 //Not more because some shocks will outright crit you, which is very unfun
+	payday_modifier = 0.75 // Matches the rest of the pay penalties the non-human crew have
+	species_language_holder = /datum/language_holder/machine
+	mutant_organs = list(/obj/item/organ/internal/cyberimp/arm/power_cord)
+	mutantbrain = /obj/item/organ/internal/brain/synth
+	mutantstomach = /obj/item/organ/internal/stomach/synth
+	mutantears = /obj/item/organ/internal/ears/synth
+	mutanttongue = /obj/item/organ/internal/tongue/synth
+	mutanteyes = /obj/item/organ/internal/eyes/synth
+	mutantlungs = /obj/item/organ/internal/lungs/synth
+	mutantheart = /obj/item/organ/internal/heart/synth
+	mutantliver = /obj/item/organ/internal/liver/synth
+	exotic_blood = /datum/reagent/fuel/oil
+	bodypart_overrides = list(
+		BODY_ZONE_HEAD = /obj/item/bodypart/head/robot/synth,
+		BODY_ZONE_CHEST = /obj/item/bodypart/chest/robot/synth,
+		BODY_ZONE_L_ARM = /obj/item/bodypart/l_arm/robot/synth,
+		BODY_ZONE_R_ARM = /obj/item/bodypart/r_arm/robot/synth,
+		BODY_ZONE_L_LEG = /obj/item/bodypart/l_leg/robot/synth,
+		BODY_ZONE_R_LEG = /obj/item/bodypart/r_leg/robot/synth,
+	)
+	digitigrade_customization = DIGITIGRADE_OPTIONAL
+	/// The innate action that synths get, if they've got a screen selected on species being set.
+	var/datum/action/innate/monitor_change/screen
+	/// This is the screen that is given to the user after they get revived. On death, their screen is temporarily set to BSOD before it turns off, hence the need for this var.
+	var/saved_screen = "Blank"
+
+/datum/species/synthetic/spec_life(mob/living/carbon/human/human)
+	if(human.stat == SOFT_CRIT || human.stat == HARD_CRIT)
+		human.adjustFireLoss(1) //Still deal some damage in case a cold environment would be preventing us from the sweet release to robot heaven
+		human.adjust_bodytemperature(13) //We're overheating!!
+		if(prob(10))
+			to_chat(human, span_warning("Alert: Critical damage taken! Cooling systems failing!"))
+			do_sparks(3, TRUE, human)
+
+/datum/species/synthetic/spec_revival(mob/living/carbon/human/transformer)
+	switch_to_screen(transformer, "Console")
+	addtimer(CALLBACK(src, .proc/switch_to_screen, transformer, saved_screen), 5 SECONDS)
+	playsound(transformer.loc, 'sound/machines/chime.ogg', 50, TRUE)
+	transformer.visible_message(span_notice("[transformer]'s [screen ? "monitor lights up" : "eyes flicker to life"]!"), span_notice("All systems nominal. You're back online!"))
+
+/datum/species/synthetic/spec_death(gibbed, mob/living/carbon/human/transformer)
+	. = ..()
+	saved_screen = screen
+	switch_to_screen(transformer, "BSOD")
+	addtimer(CALLBACK(src, .proc/switch_to_screen, transformer, "Blank"), 5 SECONDS)
```

Comment:
I feel like I possibly asked this already, but what happens to those that don't actually use screens? Do they still get a BSOD applied?

Reply:
Yea, the `switch_to_screen` proc does a null check before trying to do anything.